tests/generation/test_utils.py

```diff
@@ -3454,3 +3465,41 @@ def prepare_inputs_for_generation(self, *args, foo=False, encoder_outputs=None,
             encoder_outputs=encoder_outputs,
         )
         self.assertListEqual(outputs_assisted.tolist(), outputs_foo.tolist())
+
+    def test_return_unprocessed_logit_scores(self):
```

Comment:
I'd redo this test, to verify `output_logits`'s main property: that the logits match the forward pass. e.g. we could generate with `max_new_tokens=1, do_sample=True` and confirm that the returned logits match the forward pass :)

Reply:
@gante Thanks again for your feedback, I've implemented all of it.
I've implemented an extra unit test doing as you suggested, called test_compare_unprocessed_logit_scores().
I've also kept the original test in place, as it illustrates how one can use the output_logits option in practice.